packages/cozy-client/src/hoc.spec.jsx

```diff
@@ -82,7 +76,56 @@ describe('queryConnect', () => {
       client
     }
     const uut = mount(<WithQueries />, { context })
-    expect(uut.find(Component).prop('toto').data).toBe('hello')
-    expect(uut.find(Component).prop('tata').data).toBe('hello2')
+    expect(
+      uut
+        .find(Component)
+        .prop('simpsons')
+        .data.map(x => x.name)
+    ).toEqual(['Homer', 'Marge'])
+    expect(
+      uut
+        .find(Component)
+        .prop('upperSimpsons')
+        .data.map(x => x.name)
+    ).toEqual(['HOMER', 'MARGE'])
+  })
+})
+
+describe('queryConnectFlat', () => {
+  const setup = () => {
+    const client = setupClient()
+    const WithQueries = queryConnectFlat({
+      simpsons: queryConns.simpsons,
+      upperSimpsons: queryConns.upperSimpsons
+    })(Component)
+    return { WithQueries, client }
+  }
+
+  it('should give a display name', () => {
+    const { WithQueries } = setup()
+    expect(WithQueries.displayName).toBe('queryConnectFlat(Component)')
+  })
+
+  it('should pass result of query definitions as props', () => {
```

Comment:
Wouldn't this be true for `queryConnect` as well? would it be possible to explore the render tree to make sure there isn't more than one `<Query />` component?

Reply:
> Wouldn't this be true for queryConnect as well?

Yes this is true for queryConnect as well.

> would it be possible to explore the render tree to make sure there isn't more than one <Query /> component?

Yes I can write another test to make sure of that.

Reply:
To be clear, there will not be any `Query` component there, the data pulling from the store is done with a hook.

Reply:
https://github.com/cozy/cozy-client/pull/695/commits/e600e819ff31b1c31744cbf44f32f81086514ba6 :)